### PR TITLE
Update verify check to handle new pom files [skip ci]

### LINKED ISSUE #9650: [BUG] Github workflow for missing scala2.13 updates fails to detect when pom is new
When a PR adds a new pom that should have a corresponding scala2.13 pom, the Github workflow to check for the lack of corresponding scala2.13 updates fails to detect this case.  See #9644 as an example that was missed.

### PR DESCRIPTION
**Pull request by NVnavkumar**
Fixes #9650. 

This updates the github workflow check to properly check for newer build files that needed to be added for Scala 2.13 **and** shows the diff properly for those files and any changed files.

### CODE REVIEW

**Comment by NVnavkumar** [comment]
build

**Comment by NVnavkumar** [comment]
build

File: .github/workflows/mvn-verify-check.yml [hunk 1]
```diff
@@ -164,7 +164,8 @@ jobs:
           # verify Scala 2.13 build files
           ./build/make-scala-version-build-files.sh 2.13
           # verify git status
-          if ! git diff --exit-code 'scala2.13/*'; then
+          if [ -n "$(echo -n $(git status -s | grep 'scala2.13'))" ]; then
+              git add -N scala2.13/* && git diff 'scala2.13/*'
```
Comment (gerashegalov) [inline .github/workflows/mvn-verify-check.yml:168]: nit: * is unnecessary  for git add 

As a more compact fix we could just insert  before L167 in the current code

`git add -N scala2.13`  

which will be a NOP if there are no untracked files without changing anything else

Changes to this file don't affect the internal CI  so adding [skip ci] to the PR title to save resources.

Comment (jlowe) [inline .github/workflows/mvn-verify-check.yml:168]:
```suggestion
          if [[ -n $(git status --porcelain scala2.13) ]]; then
              git add -N scala2.13 && git diff scala2.13
```

**Review by jlowe — APPROVED** [review]
Minor simplification suggestions otherwise lgtm.

**Review by NvTimLiu — APPROVED** [review]
LGTM

**Pull request merged**